apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.html.ts

```diff
@@ -0,0 +1,563 @@
+          <input
+            type="text"
+            value="${groupMax}"
+            class="form-control"
+            id="formMax"
+            name="max_group_size"
+          />
+        </div>
+        <div class="d-flex justify-content-end">
+          <div class="form-group mb-0">
+            <input type="hidden" name="__action" value="auto_assessment_groups" />
+            <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+            <button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
+            <button type="submit" class="btn btn-primary">Group</button>
+          </div>
+        </div>
+      </form>
+    `,
+  });
+}
+
+function AddGroupModal({ csrfToken }: { csrfToken: string }) {
+  return Modal({
+    id: 'addGroupModal',
+    title: 'Add a group',
+    body: html`
+      <form name="add-group-form" method="POST">
+        <div class="modal-body">
+          <div class="form-group">
+            <label for="formName">Group Name</label>
+            <input type="text" class="form-control" id="formName" name="group_name" />
+          </div>
+          <br />
+          <div class="form-group">
+            <label for="formUids">UIDs</label>
+            <input
+              type="text"
+              class="form-control"
+              id="formUids"
+              name="uids"
+              placeholder="one@example.com, two@example.com, three@example.com"
+            />
+            <small id="uidHelp" class="form-text text-muted">
+              Separate with "," <br />
+              Please make sure they are not in any other groups
+            </small>
+          </div>
+        </div>
+      </form>
+    `,
+    footer: html`
+        <input type="hidden" name="__action" value="add_group" />
+        <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+        <button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
+        <button type="submit" class="btn btn-primary">Add</button>
+        </form>
+    `,
+  });
+}
+
+function DeleteGroupModal({
+  csrfToken,
+  assessmentSetName,
+  assessmentNumber,
+}: {
+  csrfToken: string;
+  assessmentSetName: string;
+  assessmentNumber: number;
+}) {
+  return Modal({
+    id: 'deleteAllGroupsModal',
+    title: 'Delete all existing groups',
+    body: html`
+      <p>
+        Are you sure you want to delete all existing groups for
+        <strong>${assessmentSetName} ${assessmentNumber}</strong>? This cannot be undone.
+      </p>
+    `,
+    footer: html`
+      <form name="delete-all-form" method="POST">
+        <input type="hidden" name="__action" value="delete_all" />
+        <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+        <button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
+        <button type="submit" class="btn btn-danger">Delete all</button>
+      </form>
```

Comment:
This is a bit of an idiosyncratic thing we do with our modals, but the `Modal` component actually itself already includes a form:

https://github.com/PrairieLearn/PrairieLearn/blob/342a3df058526d26022c37735e6329d8ab52a11b/apps/prairielearn/src/components/Modal.html.ts#L32-L37

This is because we often end up in situations where we need a form that spans both the modal body and footer. I'd recommend relying on that built-in form here.

Reply:
That explains why I failed CI for issues regarding the CSV upload. I have now updated all modals here to use the built-in form from the modal component in [dc8a69d](https://github.com/PrairieLearn/PrairieLearn/pull/9550/commits/dc8a69d5bba27345198abea501037c2deb732050).